drew/event_emitter.js

```diff
@@ -0,0 +1,51 @@
+'use strict'
+const fs = require('fs');
+const ee = new (require('events').EventEmitter)
+
+var reader = module.exports = function(cb) {
+  let files = [];
+  ee.on('file-one-read', (data) => {
+    fs.readFile('./two.txt', (err,data) => {
+      ee.emit('file-two-read', data)
+    })
+    var oneData = data.toString();
+    var bufferOneData = new Buffer(oneData);
+    var bufferOneDataStringed = bufferOneData.toString('hex');
+    var bufferOneDataStringedSliced = bufferOneDataStringed.slice(0,8);
+    console.log('the first 8 bytes of file one read: ', bufferOneDataStringedSliced);
+    files.push(bufferOneDataStringedSliced)
+  })
+
+  fs.readFile('./one.txt', (err,data) => {
+    console.log(data);
+    ee.emit('file-one-read', data)
+  })
+
+  ee.on('file-two-read', (data) => {
+    fs.readFile('./three.txt', (err,data) => {
+      ee.emit('file-three-read', data)
+    })
+    var twoData = data.toString();
+    var bufferTwoData = new Buffer(twoData);
+    var bufferTwoDataStringed = bufferTwoData.toString('hex');
+    var bufferTwoDataStringedSliced = bufferTwoDataStringed.slice(0,8);
+    console.log('the first 8 bytes of file two read: ', bufferTwoDataStringedSliced);
+    files.push(bufferTwoDataStringedSliced)
+  })
+
+  ee.on('file-three-read', (data) => {
+    var threeData = data.toString();
+    var bufferThreeData = new Buffer(threeData);
+    var bufferThreeDataStringed = bufferThreeData.toString('hex');
+    var bufferThreeDataStringedSliced = bufferThreeDataStringed.slice(0,8);
+    console.log('the first 8 bytes of file three read: ', bufferThreeDataStringedSliced);
+    files.push(bufferThreeDataStringedSliced)
+    cb(files);
+  })
+
+}
+
```

Comment:
It looks like you have a good grasp of how the fs module and the event emitters work.  You might have a little extra code in here that could use some refactoring.  Also, you are not handling the asynchronous nature of the filereads well.  You are always going to get file one logging first.  But the way you have this setup there is no guarantee that file three will not log out before file two.  See if you can diagnose this and refactor it a little to fix this issue.